contrib/exporters/core/extend.go

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package core
+
+import (
+	"strings"
+	"encoding/json"
+	"github.com/spf13/viper"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/api/client"
+)
+
+// for no transform extensions
+type extendNone struct {
+}
+
+func NewExtendNone(cfg *viper.Viper) (interface{}, error) {
+	logging.GetLogger().Infof("entering NewExtendNone")
+        return &extendNone{}, nil
+}
+
+func (e *extendNone)  Extend(in interface{}) interface{} {
+	return in
+}
+
+// for transform extensions
+// Syntax supported is:
+//	VAR_NAME=<gremlin expression with a single substitution string>
+// For example:
+//	AA_Name=G.V().Has('RoutingTables.Src',%Network.A%).Values('Host')
+// Substitution string must be surrounded with `%`
+// The substitution string refers to fields that exist in the data provided by the particular transform
+// No spaces allowed
+// Need to define `extend` field as a subfield of `transform`
+// Structure in yml file is as follows:
+//   transform:
+//     type: <xxx>
+//     extend:
+//       - VAR_NAME1=<gremlin expression with a single substitution string>
+//       - VAR_NAME2=<gremlin expression with a single substitution string>
+type extendGremlin struct {
+	extendStrings []string
+	gremlinClient *client.GremlinQueryHelper
+}
+
+func NewExtendGremlin(cfg *viper.Viper) (interface{}, error) {
+	logging.GetLogger().Infof("entering NewExtendGremlin")
+	gremlinClient := client.NewGremlinQueryHelper(CfgAuthOpts(cfg))
+	extend := cfg.GetStringSlice(CfgRoot + "transform.extend")
+	logging.GetLogger().Infof("NewExtendGremlin, extend = %s", extend)
+        p :=  &extendGremlin{
+		extendStrings: extend,
+		gremlinClient: gremlinClient,
+	}
+	return p, nil
+}
+
+// use the substitution string to extract the value from the provided map
+func (e *extendGremlin) DeriveValue(S map[string]interface{}, varString string) string {
+        substrings := strings.Split(varString, ".")
+        var ii map[string]interface{}
+        var jj map[string]interface{}
+        ii = S
+        for i := 0; i < len(substrings)-1; i++ {
+                str := substrings[i]
+                jj = ii[str].(map[string]interface{})
+                ii = jj
+        }
+	// TODO: verify that each substring is a valid field so we don't get an exception
+        name := ii[substrings[len(substrings)-1]].(string)
+        return name
+}
+
+// use substitution value and perform gremlin query to obtain the desired data enrichment
+func (e *extendGremlin) PerformConversion(origString string, S map[string]interface{}) string {
+	// `%` marks the substitutin string; split the string on `%`
+        substrings := strings.Split(origString, "%")
```

Comment:
this will not work for the general case of multiple substition values. You must use some type of template library such as: https://github.com/Masterminds/sprig

Reply:
Is this a requirement to have multiple substitutions in a single query? We can expand the functionality over time to cover this case, if needed.

Reply:
Is this a requirement to have multiple substitutions in a single query? We can expand the functionality over time to cover this case, if needed.

Reply:
its' mandatory to be able to express the current resolvers (e.g. runC) using your queries. And as the runC requires both tidNode and IP you must support these. More generally using split according to `%` to replace even a single argument is a very bad idea as it does not account `%` which may be legitimately embedded into the query.

Reply:
We now use templates, so this should be OK.